pyesbulk/__init__.py

```diff
@@ -183,25 +281,74 @@ def actions_tracking_closure(cl_actions):
                     # ... otherwise consider it successful.
                     successes += 1
             elif status == 400:
-                doc = {
+                try:
+                    exc_payload = resp["exception"]
+                except KeyError:
+                    pass
+                else:
+                    resp["exception"] = repr(exc_payload)
+                jsonstr = json.dumps(
+                    {
                         "action": action,
                         "ok": ok,
                         "resp": resp,
                         "retry_count": retry_count,
-                        "timestamp": tstos(time.time())
-                        }
-                jsonstr = json.dumps(doc, indent=4, sort_keys=True)
+                        "timestamp": _tstos(),
+                    },
+                    indent=4,
+                    sort_keys=True,
+                )
```

Comment:
Random JSON dumps in the log file are hard to interpret; it'd be nice to have a bit of context, too.

Reply:
Actually, these are not really random JSON dumps.  They contain the response object, and the original "action" dictionary that was sent.  The intent is to have this captured in a file, not sent in some log message, so that the caller can decide what to do with actions that were indexed.

Most of the time, the response object has enough context, combined with the original action payload, so that one can do something about it.

Reply:
My point was that you're writing a JSON dump into a file without any situation/location context. Someone familiar with the code may recognize where that would have been written and why, but seeing just the logged output would give anyone else no clue at all what it means.